src/org/jitsi/impl/neomedia/rtcp/RTCPHeaderUtils.java

```diff
@@ -0,0 +1,138 @@
+            return -1;
+        }
+
+        return buf[off + 1] & 0xff;
+    }
+
+    /**
+     * Gets the RTCP sender SSRC.
+     *
+     * @param buf the byte buffer that contains the RTCP header.
+     * @param off the offset in the byte buffer where the RTCP header starts.
+     * @param len the number of bytes in buffer which constitute the actual
+     * data.
+     * @return the unsigned RTCP sender SSRC, or -1 in case of an error.
+     */
+    public static long getSenderSSRC(byte[] buf, int off, int len)
+    {
+        if (buf == null ||  Math.min(buf.length, len) < off + 8)
+        {
+            return -1;
+        }
+
+        return (((buf[off + 4] & 0xff) << 24)
+            | ((buf[off + 5] & 0xff) << 16)
+            | ((buf[off + 6] & 0xff) << 8)
+            | (buf[off + 7] & 0xff)) & 0xffffffffl;
+    }
+
+    /**
+     * Gets the RTCP packet length in bytes.
+     *
+     * @param buf the byte buffer that contains the RTCP header.
+     * @param off the offset in the byte buffer where the RTCP header starts.
+     * @param len the number of bytes in buffer which constitute the actual
+     * data.
+     * @return  the RTCP packet length in bytes, or -1 in case of an error.
+     */
+    public static int getLength(byte[] buf, int off, int len)
+    {
+        if (buf == null ||  Math.min(buf.length, len) < off + 4)
+        {
+            return -1;
+        }
+
+        int lengthInWords
+            = ((buf[off + 2] & 0xff) << 8) | (buf[off + 3] & 0xff);
+
+        return (lengthInWords + 1) * 4;
+    }
+
+    /**
+     * Gets the RTCP packet version.
+     *
+     * @param buf the byte buffer that contains the RTCP header.
+     * @param off the offset in the byte buffer where the RTCP header starts.
+     * @param len the number of bytes in buffer which constitute the actual
+     * data.
+     * @return the RTCP packet version, or -1 in case of an error.
+     */
+    public static int getVersion(byte[] buf, int off, int len)
+    {
+        if (buf == null ||  Math.min(buf.length, len) < off + 1)
+        {
+            return -1;
+        }
+
+        return (buf[off] & 0xc0) >>> 6;
+    }
+
+    /**
+     * Checks whether the RTCP header is valid or not. It does so by checking
+     * the RTCP header version and makes sure the buffer is at least 8 bytes
+     * long.
+     *
+     * @param buf the byte buffer that contains the RTCP header.
+     * @param off the offset in the byte buffer where the RTCP header starts.
+     * @param len the number of bytes in buffer which constitute the actual
+     * data.
+     * @return true if the RTCP packet is valid, false otherwise.
+     */
+    public static boolean isValid(byte[] buf, int off, int len)
+    {
+        int version = RTCPHeaderUtils.getVersion(buf, off, len);
+        if (version != RTCPHeader.VERSION)
+        {
+            return false;
+        }
+
+        int pktLen = RTCPHeaderUtils.getLength(buf, off, len);
+        if (pktLen < RTCPHeader.SIZE)
```

Comment:
getVersion checks whether the length is at least 1 byte; getLength checks whether the length is at least 4 bytes. There is no check that what getLength returns is:
a) <= len
b) <= buf.length - off